src/biosequence/indexing.jl

```diff
@@ -8,7 +8,7 @@
 ### License is MIT: https://github.com/BioJulia/BioSequences.jl/blob/master/LICENSE.md
 
 @inline function Base.iterate(seq::BioSequence, i::Int = firstindex(seq))
-    i > lastindex(seq) ? nothing : @inbounds seq[i], i + 1
+    (i % UInt) - 1 < lastindex(seq) ? (@inbounds seq[i], i + 1) : nothing
```

Comment:
```suggestion
    (i % UInt) - 1 < (lastindex(seq) % UInt) ? (@inbounds seq[i], i + 1) : nothing
```

Reply:
The UInt/Int comparison adds an extra branch. Don't know how arrays avoids this, but we can elide it like this